Release 0.8.0

marcelklehr opened the pull request:
The first release candidate for v0.8.0 is ready for testing!

Download the installer at https://github.com/marcelklehr/nodist/releases/tag/v0.8.0-rc1

marcelklehr committed:
v0.8.0-rc1

marcelklehr committed:
[cli] Update help output

marcelklehr committed:
[installer] Don't run selfupdate on install

-- it was removed.

marcelklehr committed:
[installer] Fix #99: Remove npm prefix on uninstall

marcelklehr committed:
npm shim + global/local/env switching

- Introduced .npm-version and NODIST_NPM_VERSION
- Renamed NODIST_VERSION to NODIST_NODE_VERSION
- added npm shim
- Factored out common function into nodist.go
- added subcommands for npm
- added new functionality to lib/npm.js
- adjust build script

marcelklehr committed:
[shim] Implement npm match + [nodist] Save version list to disk

marcelklehr committed:
[build] Fix npm version

marcelklehr committed:
[isntaller] Don't show components page

marcelklehr committed:
[installer] Fix x64 detection

marcelklehr committed:
[installer] Fix setting npm prefix

gmahomarf committed:
Added nodist wrapper function and modified install script for bash

marcelklehr committed:
[installer] Fix x64 detection, finally

marcelklehr committed:
[mingw] Fix bash script

gmahomarf committed:
This should make it work in Cygwin (sans env and use capabilites)

gmahomarf commented on bin/nodist:
```diff
@@ -1,10 +1,16 @@
\ No newline at end of file
+    DIR=$(echo "$NODIST_PREFIX" | sed -e 's,\\,/,g')
+    "$DIR/bin/node.exe" "$DIR/cli" "$@"
+fi
```
You should remove everything from this file, except lines 14 and 15, then move the npm stuff to the nodist() function in nodist.sh

marcelklehr replied:
Crap, you're right... it's late _facepalm_

marcelklehr committed:
Merge #136: Added wrappers and modified install script for MinGW bash

gmahomarf commented on bin/nodist.sh:
```diff
@@ -0,0 +1,38 @@
+            echo $res
+        fi
+    else
+    if [ "$1" = "npm" -o "$2" = "env" ]; then
```
~~Should be -a not -o~~ Better yet, do

``` bash
elif [ "$1" = "npm" -a \( "$2" = "env" -o "$2" = "use" \) ]; then
```

To allow for both `env` and `use`

gmahomarf replied:
~~Should also probably be `elif` instead of `else if`~~ see above

marcelklehr replied:
I decided not to introduce `use` to npm version switching since it's going to be deprecated in 0.8.0, anyway

gmahomarf replied:
Ah, I was not aware. Never mind then. Removed my other comment.

gmahomarf commented on bin/nodist.sh:
```diff
@@ -0,0 +1,38 @@
+            echo "Please specify a version to use."
+            ret=1
+        else
+            res=`"$NODIST_BIN_DIR__/nodist" + "$2"`
```
Missing "npm" argument

marcelklehr committed:
Fix build + Fix fetch on 404 + use official node-progress

+ clean up nsis script
+ cp -r unzipped npm instead of moving it (unzip probably doesn't realease its handle, so we just let it cling to it)
+ if nodejs.org/dist status is not 200 we bail
+ using the official node-progress has the advantage of bugfixes

marcelklehr committed:
[README] Add NSIS plugin to build requirements

marcelklehr committed:
Fix fetch: Don't throw on 404 + Don't try to display down speed

marcelklehr committed:
Fix `nodist env`: Fix env var in cmd script and cli.js

marcelklehr committed:
Fix progress bar

wasn't being initialized because of a typo
fixes #139

marcelklehr committed:
Fix npmist install + Replace unzip with extract-zip + Pin node-progress

fixes #133

marcelklehr committed:
Fix build: rename global version files to .node-version-global

+ fix shim-node to use the correct env var

marcelklehr committed:
Fix npmist: Use new global version filename

marcelklehr committed:
Fix npmist.install: repoPath not defined

marcelklehr committed:
Use node-tar to unzip npm archive from github

marcelklehr committed:
[build]+[readme] Create chocolatey package + add choco install to readme

marcelklehr committed:
[readme] Add migration note about changed behaviour

+ add chocolatey package to changelog
